docs/static/performance-checklist.asciidoc

```diff
@@ -49,10 +49,43 @@ sure-fire way to create a confusing situation.
+
+[[performance-troubleshooting-worker-utilization]]
+===== Worker utilization
+
+When a pipeline's `worker_utilization` flow metric is consistently near 100, all of its workers are occupied processing the filters and outputs of the pipeline.
+When this is the case, we can see _which_ plugins in the pipeline are consuming the available worker capacity by looking at the plugin-level `worker_utilization` and `worker_millis_per_event` flow metrics.
+Using this information, we can gain intuition about how to tune the pipeline's settings to add resources, or how to find and eliminate wasteful computation.
+
+In general, plugins fit into one of two categories:
+
+- *CPU-bound*: plugins that perform computation on the contents of events _without_ the use of the network or disk IO tend to benefit from incrementally increasing `pipeline.workers` as long as the process has available CPU; once CPU is exhausted additional concurrency can result in _lower_ throughput as the pipeline workers contend for resources.
+- *IO-bound*: plugins that use the network to either enrich events or transmit events tend to benefit from incrementally increasing `pipeline.workers` and/or tuning the pipeline batching as long as those external services are not exerting back-pressure (even if Logstash is using nearly all of its available CPU).
+
+When a pipeline's `worker_utilization` is _below_ 100, its pipeline workers are spending their time waiting for input.
+Because the volume of data in most pipelines is expected to ebb and flow, the goal should be to tune the pipeline such that it has the resources to avoid propagating back-pressure to its inputs during peak periods.
+
+[[performance-troubleshooting-queue-backpressure]]
+===== Queue back-pressure
+
+When a pipeline receives events faster than it can process them, the inputs eventually experience back-pressure that prevents them from receiving additional events.
+Depending on the input plugin being used, back-pressure can either propagate upstream or result in the risk of data-loss.
+
+A pipeline's `queue_backpressure` flow metric reflects how much time the inputs are spending attempting to push events into the queue.
+The metric isn't precisely comparable across pipelines, but instead allows you to compare a single pipeline's current behaviour to _itself_ over time.
+When this metric is growing, we generally need to look _downstream_ at the pipeline's filters and outputs to see if they are using resources effectively, have sufficient resources allocated, or are experiencing back-pressure of their own.
+
+NOTE: A persisted queue offers durability guarantees and can absorb back-pressure for longer than the default in-memory queue, but once it is full it too propagates back-pressure.
+      The `queue_persisted_growth_events` flow metric is useful measure of how much back-pressure is being actively absorbed by the persisted queue, and should trend toward zero (or less) over the pipeline's lifetime.
 
 [[tuning-logstash]]
```

Comment:
It is not clear to me why this section-break results in a separate page. I had hoped that the bits above (the why) would be more closely-tied to the bits below (the how).

Is it just the heading level, or is there something else?

Reply:
Is the presentation you want similar to this?  If so, the `[float]` tag is the magic you're looking for. 

<img width="1105" alt="Screen Shot 2024-06-27 at 4 58 59 PM" src="https://github.com/elastic/logstash/assets/35154725/a8c04b03-2763-46c1-a545-7e25f9918044">